src/herder/TxSetFrame.cpp

```diff
@@ -512,6 +513,7 @@ TxSetFrame::getTotalFees(LedgerHeader const& lh) const
 void
 TxSetFrame::toXDR(TransactionSet& txSet)
 {
+    sortForHash();
```

Comment:
It is because `TxSetFrame` and friends are a mess. `AbstractTxSetFrame` (which is all we have a pointer to in `LedgerManagerImpl`) does not have a `sortForHash` method, and shouldn't because `SimulationTxSetFrame` has only `const` members (this is a good design I want to keep).

In contrast, `SimulationTxSetFrame::toXDR` actually constructs, sorts-for-hash, and calls `toXDR` on a temporary `TxSet`, which it then throws out. I could theoretically do something like this in `TxSetFrame::toXDR` as well -- at least the construction-and-sorting part, the delegation has to bottom out with an xdrpp call somewhere! -- and remove or change the other call sites. Not sure what you prefer.

I should also point out that the not-very-well-named `TxSetFrame::sortForApply` doesn't actually sort-in-place either: it produces a new sorted vector for application. I'd be happy to try moving the `sortForHash` interface to this sort of shape too, but I wasn't sure how much deeper the pile of yak hair should get in this PR. Happy to go deeper if you like.

Basically `TxSetFrame` is a mutable/stateful type that's half way through migrating to an immutable/stateless type. I was trying to fit into that existing situation without knocking too many other things over.

Reply:
fixed to just assert that it's sorted